Spigot-API-Patches/0006-Adventure-Events.patch

```diff
@@ -0,0 +1,1062 @@
++    }
++
++    /**
++     * Sets the reason why the player is getting kicked
++     *
++     * @param kickReason kick reason
++     */
++    public void reason(@org.jetbrains.annotations.Nullable net.kyori.adventure.text.Component kickReason) {
++        this.kickReason = kickReason;
++    }
++    // Paper end
++
++    /**
++     * Gets the reason why the player is getting kicked
++     *
++     * @return string kick reason
++     * @deprecated in favour of {@link #reason()}
++     */
+     @NotNull
++    @Deprecated // Paper
+     public String getReason() {
+-        return kickReason;
++        return net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().serialize(this.kickReason); // Paper
+     }
+ 
+     /**
+      * Gets the leave message send to all online players
+      *
+      * @return string kick reason
++     * @deprecated in favour of {@link #leaveMessage()}
+      */
+     @NotNull
++    @Deprecated // Paper
+     public String getLeaveMessage() {
+-        return leaveMessage;
++        return net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().serialize(this.leaveMessage); // Paper
+     }
+ 
+     @Override
+@@ -55,18 +104,22 @@ public class PlayerKickEvent extends PlayerEvent implements Cancellable {
+      * Sets the reason why the player is getting kicked
+      *
+      * @param kickReason kick reason
++     * @deprecated in favour of {@link #reason(net.kyori.adventure.text.Component)}
+      */
++    @Deprecated // Paper
+     public void setReason(@NotNull String kickReason) {
+-        this.kickReason = kickReason;
++        this.kickReason = net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserialize(kickReason); // Paper
+     }
+ 
+     /**
+      * Sets the leave message send to all online players
+      *
+      * @param leaveMessage leave message
++     * @deprecated in favour of {@link #leaveMessage(net.kyori.adventure.text.Component)}
+      */
++    @Deprecated // Paper
+     public void setLeaveMessage(@NotNull String leaveMessage) {
+-        this.leaveMessage = leaveMessage;
++        this.leaveMessage = net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserialize(leaveMessage); // Paper
+     }
+ 
+     @NotNull
+diff --git a/src/main/java/org/bukkit/event/player/PlayerLocaleChangeEvent.java b/src/main/java/org/bukkit/event/player/PlayerLocaleChangeEvent.java
+index 1db386bb701cb6974daedc6bb5b93a3afbc42100..84521186404b8e43c81a2f9513dce2be40d27840 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerLocaleChangeEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerLocaleChangeEvent.java
+@@ -12,18 +12,32 @@ public class PlayerLocaleChangeEvent extends PlayerEvent {
+     private static final HandlerList handlers = new HandlerList();
+     //
+     private final String locale;
++    // Paper start
++    private final java.util.Locale adventure$locale;
```

Comment:
The dollar sign should be used only in mechanically generated source code or, rarely, to access pre-existing names on legacy systems.